api/v1alpha1/queuedworkload_types.go

```diff
@@ -39,6 +39,13 @@ type QueuedWorkloadSpec struct {
 
 	// admission holds the parameters of the admission of the workload by a ClusterQueue.
 	Admission *Admission `json:"admission,omitempty"`
+
+	// priority determines the order of access to the resources in the
+	// ClusterQueue associated with the queue where the workload is located.
+	// The higher the value, the higher the priority.
+	// It's ClusterQueue-scoped and generated according to the priorityclass of the
+	// job.
```

Comment:
how will priority impact order exactly? will workloads always be ordered first by priority then the QueueingStrategy? or will itself be one of the queuing strategies?

Reply:
you got me here, I was also looking for the func re-sorting every time a new queuedWorkload was added to a queue

Reply:
> how will priority impact order exactly? will workloads always be ordered first by priority then the QueueingStrategy? or will priority itself be one of the queuing strategies?

Yes. It is one of the queuing strategies. Probably similar to yarn, where the user specifies whether to sort workloads by priority.  The workloads are ordered by the priority in the ClusterQueue.

> I was also looking for the func re-sorting every time a new queuedWorkload was added to a queue

It's only api definition. The implementation of the `QueueingStrategy` depends on the https://github.com/kubernetes-sigs/kueue/pull/97

Reply:
We have to think of 2 things: the default strategy and options for configuration.

I think the default strategy should be order by priority and then by creation time in the case of ties.

As for configuration, we can defer that to #8 and #10.

Reply:
> I think the default strategy should be order by priority and then by creation time in the case of ties.

where does OrderStrategy fit here? priority then OrderStrategy (which would be creation time or re-queue depending on the strategy)?